pkg/file/resolution.go

```diff
@@ -0,0 +1,158 @@
+package file
+
+import (
+	"sort"
+
+	"github.com/scylladb/go-set/strset"
+)
+
+// Resolution represents the fetching of a possibly non-existent file via a request path.
+type Resolution struct {
+	RequestPath Path
+	*Reference
```

Comment:
Is reference `Reference` what is fetched here? Is this why we use a pointer since it can be non-existent?

Reply:
exactly 👍